dask/dataframe/io/tests/test_parquet.py

```diff
@@ -733,7 +733,7 @@ def test_append_wo_index(tmpdir, engine):
     assert_eq(df.set_index("f"), ddf3)
 
 
-def test_append_overlapping_divisions(tmpdir):
+def test_append_overlapping_divisions(tmpdir, engine):
```

Comment:
So this is the critical change - before it would try with the default engine, and error when there were none, but now it skips if one or other engine is missing, correct?

Reply:
Correct. The test is now run twice, once with `fastparquet` as `engine` and another time with `pyarrow`. In both cases, the test will be skipped if the corresponding `engine` is not installed.